### DISCUSSION TITLE
Create relation o2a

### DISCUSSION THREAD

**Discussion by cedricDevWP** [directus/directus#14697] [Feature Requests]
Hi,

I think it would be wise to add an o2a relationship, for example I created a singleton which has 1 single object of a custom type except that it is impossible for me to make an o2a relationship. I have to make an m2a relation and add a count validation equal to 1. I find this overkill compared to just a new column with the primary key of my linked table.

Moreover, if I redo this on a custom type, it displays this to me at the level of the list:

![Capture d’écran 2022-07-27 à 13 57 57](https://user-images.githubusercontent.com/12508260/181241275-a6ea60d4-a7cf-4104-845e-8faa55f77b8e.png)
 
and suddenly it's not speaking...

I tried to find an extension that allows you to do o2a but I couldn't find it. I also tried to create this relationship by following this page https://docs.directus.io/extensions/creating-extensions/ but it's quite complex I find.

maybe i missed something...if anyone has any idea :)

thx

**Comment by rijkvanzanten** [comment]
Technically, the API already supports this type[^1] of relationship, as the current "m2a" is technically nothing more than a `o2m -> a2o` combination. That being said, the app interface for that type of relationship doesn't currently exist. We would have to create an interface type that edits two field-values simultaneously, but render it as a single field. That's not something that the app supports at this point (for custom interfaces or otherwise)

[^1]: We call it `a2o` internally, though the idea is the same

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
